### PR TITLE
Use actual vendor dir

### PR DESCRIPTION
**Pull request by dantleech**
This is another approach to #460 which avoids the global state dependency.

To be honest, the whole bootstrapping process needs to be refactored and probably migrated to use the Symfony DI container and Configuration, as with #405

### CODE REVIEW

File: lib/Container/PhpactorContainer.php [hunk 1]
```diff
@@ -85,7 +85,7 @@ public function register(string $serviceId, Closure $factory, array $tags = [])
 
     public function getParameter(string $name)
     {
-        if (!isset($this->parameters[$name])) {
+        if (!array_key_exists($name, $this->parameters)) {
```
Comment (dantleech) [inline lib/Container/PhpactorContainer.php]: Add a test for this

File: lib/Phpactor.php [hunk 2]
```diff
@@ -53,12 +53,16 @@ public static function boot(InputInterface $input): PhpactorContainer
...
-        $container->register('config.paths', function () {
-            return new Paths();
+        $paths = new Paths();
+        $container->register('config.paths', function () use ($paths) {
+            return $paths;
         });
 
+        if (!isset($config['cache_dir'])) {
+            $config['cache_dir'] = $paths->userCacheDir();
+        }
```
Comment (dantleech) [inline lib/Phpactor.php]: Argh.

**Comment by talbergs** [comment]
That's awesome, can we merge it?
Thou one minor thing I noticed - https://cl.ly/2H3U1n0b2x1H
Of course I do not need this configurable, just noticed...

**Comment by dantleech** [comment]
> Of course I do not need this configurable,

Yeah, it's an immutable parameter, but I wanted to keep it in the "config" so that it can be inspected. Ideally should be handled better, but will hopefully change all of this at some point.

**Pull request merged**